### PR TITLE
fix - removed annoying message 'enter your password for the ssh key'

### PR DESCRIPTION
**Pull request by freedev**
On OSX 10.10.5 I am unable to ssh into VMs executing this command:

```
ssh -F  ./output/kube_da8d7eb84e2ecc_ssh_conf kube-00
```

A popup window asking the passphrase appears everytime I try to enter into the VMs.

screen shot 2015-08-29 at 13 02 18

This fix remove the empty passphrase from the key.

### CODE REVIEW

**Comment by freedev** [comment]
@smarterclayton I see some checks were unsuccessful. was the fix accepted?

**Comment by brendandburns** [comment]
re-triggered shippable, I think it was a flake.

@jeffmendoza for his thoughts on the azure bits

File: docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js [hunk 1]
```diff
@@ -107,6 +107,9 @@ var create_ssh_key = function (prefix) {
   };
   openssl.exec('req', opts, function (err, buffer) {
     if (err) console.log(clr.red(err));
+    openssl.exec('rsa', { in: opts.keyout, out: opts.keyout }, function (err, buffer) {
+       if (err) console.log(clr.red(err));
+    });
     fs.chmod(opts.keyout, '0600', function (err) {
```
Comment (jeffmendoza) [inline docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js]: I'm not a js programmer, but wouldn't the chmod need to go inside the callback for the second exec to operate on the correct file?

Can you verify that the file permissions are correct on the generated key after this change?

Adding @errordeveloper

Comment (freedev) [inline docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js]: @jeffmendoza I thought about this fix. 
openssl rsa line rewrites the key removing the phassphrase. It is optional and should be executed to fix a strange behaviour happening on OSX, even if not works correctly it doesn't mean that the entire process should be stopped.
On the other hand fs.chmod line is mandatory.
Just to be more precise, I could change the openssl error message into a warning.

Comment (jeffmendoza) [inline docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js]: Isn't it possible that the chmod and 'openssl rsa' happen in parallel and the 'openssl rsa' command clobbers the permission change, leaving the file world readable?

Comment (freedev) [inline docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js]: @jeffmendoza I got your point. Let me double check please.

**Comment by brendandburns** [comment]
@freedev is this now ready?

**Comment by freedev** [comment]
@brendandburns sorry I was quite busy in the latest days. As far I see looking at openssl.exec implementation, the "rsa" command is executed using "spawn" and is an asynchronous process creation. So as @jeffmendoza states, we cannot be sure about what the openssl rsa command do if it is executed later. Just to be sure that everything go well, I have moved the fs.chmod inside the callback.

**Comment by brendandburns** [comment]
@jeffmendoza can you give this a once over?

**Comment by jeffmendoza** [comment]
LGTM
Thanks!

**Comment by a-robinson** [comment]
Hi @freedev - could you please apply your change as a single commit on top of our master branch, rather than as two commits with a bunch of merges smashed in the middle?

Just as a general tip, it's preferable to use `git rebase upstream/master` rather than `git merge upstream/master` to avoid building up all those merge commits that we don't want.

**Pull request merged**